tests/system/Database/Builder/WhereTest.php

```diff
@@ -446,4 +449,119 @@ public function testWhereWithLower()
         $expectedSQL = 'SELECT * FROM "jobs" WHERE LOWER(jobs.name) = \'accountant\'';
         $this->assertSame($expectedSQL, str_replace("\n", ' ', $builder->getCompiledSelect()));
     }
+
+    public function testWhereValueIsString()
+    {
+        $builder = $this->db->table('users');
+
+        $builder->where('id', '1');
+
+        $expectedSQL = <<<'SQL'
+            SELECT * FROM "users" WHERE "id" = '1'
+            SQL;
+        $this->assertSame($expectedSQL, str_replace("\n", ' ', $builder->getCompiledSelect()));
+    }
+
+    public function testWhereValueIsFloat()
+    {
+        $builder = $this->db->table('users');
+
+        $builder->where('id', 1.234);
+
+        $expectedSQL = <<<'SQL'
+            SELECT * FROM "users" WHERE "id" = 1.234
+            SQL;
+        $this->assertSame($expectedSQL, str_replace("\n", ' ', $builder->getCompiledSelect()));
+    }
+
+    public function testWhereValueIsTrue()
+    {
+        $builder = $this->db->table('users');
+
+        $builder->where('id', true);
+
+        $expectedSQL = 'SELECT * FROM "users" WHERE "id" = 1';
```

Comment:
Is this the expected result?

Reply:
Yes, MySQL has no support for boolean values so they're converted to `0` or `1`.

Reply:
What about other databases?